packages/frontend/src/contexts/LayoutNavigation.tsx

```diff
@@ -0,0 +1,29 @@
+import { createContext, ReactNode } from 'react'
+import { DrawerLink } from 'components/Layout'
+
+export type LayoutNavigationProviderData = {
+  links: DrawerLink[]
+  isDrawerOpen: boolean
+  openDrawer: () => void
+  closeDrawer: () => void
+}
+
+interface LayoutNavigationProviderProps {
+  children: ReactNode
+  value: LayoutNavigationProviderData
+}
+
+export const LayoutNavigationContext = createContext(
+  {} as LayoutNavigationProviderData,
+)
+
+export function LayoutNavigationProvider(
```

Comment:
Don't think we need this if we hoist the sidebar components out

Reply:
Hmm initially i wanted to hoist it out but stacey wanted the drawer to be right beside the flows/apps/executions title which means I felt that it has to be part of the Flows page itself? we can discuss about this later